src/amo/components/AddonReviewList/index.js

```diff
@@ -54,22 +57,30 @@ type InternalProps = {|
   clientApp: string,
   dispatch: DispatchFunc,
   errorHandler: ErrorHandlerType,
+  featuredReview?: UserReviewType,
+  history: ReactRouterHistoryType,
   i18n: I18nType,
   lang: string,
+  loadingReview: boolean,
   match: {|
     ...ReactRouterMatchType,
     params: {
-      addonSlug: string,
+      addonSlug?: string,
+      reviewId?: number,
     },
   |},
   pageSize: number | null,
   reviewCount?: number,
   reviews?: Array<UserReviewType>,
-  history: ReactRouterHistoryType,
+  reviewsAreLoading: boolean,
+  siteUser: UserType | null,
 |};
 
 export class AddonReviewListBase extends React.Component<InternalProps> {
-  componentWillMount() {
+  // TODO: Note: I had to change this from componentWillMount to componenDidMount
+  // otherwise this patch breaks with SSR.
+  // I know this is not ideal, but I'm not sure how else to deal with it at this time.
```

Comment:
Could you explain how it breaks? I'm pretty sure we need to keep rendering reviews on the server for SEO.

Reply:
The page never finishes loading. It's left with loading indicators for each of the areas. I encountered this before, and the explanation, as I understood it from @willdurand was that the sagas that are dispatched on the server do not complete in time, so the fetches are dispatched, but the loads never occur.

I agree that we should endeavour to keep rendering the reviews on the server, but this also needs to be addressed. Maybe moving it into its own component, as you suggest below, might help. It might be okay to not render the reviews on the server just for the case of a featuredReview permalink destination.

Reply:
The way our SSR render cycle works is that the component has to dispatch *all* data loading actions in the *first* render. After that, the server will wait for all running sagas to complete and then re-render the page. This process is fragile but it should be possible to achieve SSR, theoretically.

Reply:
I think the problem comes because we now have a case where we need to fetch a review first (via id), before we can fetch an add-on or a list of reviews, because only after fetching the review do we know the add-on's slug. I suppose we could just add the slug as part of the link and always require it - but we could encounter the case where a review that was written at one point is for an add-on whose slug has changed, and that would cause a problem.

Reply:
I resolved this issue by always requiring an `addonSlug` parameter.